http_101.rst

```diff
@@ -1,20 +1,369 @@
+503	      Service Unavailable    The server is currently unable to fulfill the request.
+============= =====================  ====================
+
+
+
 Tools: Speaking http with telnet/netcat/curl
 ============================================
 
-Understanding how a browser/tool parses a url and server receives it for a
-request
+Telnet
+------
+
+The telnet utility is used for interactive communication to a host on a given port. 
+Once the connection to the remote host is established, an HTTP request can be send to the host by typing it in the prompt:
+
+.. code-block:: console
+
+	$ telnet apache.org 80
+	Trying 88.198.26.2...
+	Connected to apache.org.
+	Escape character is '^]'.
+	GET /index.html HTTP/1.1
+	Host: apache.org
+
+	HTTP/1.1 200 OK
+	Date: Wed, 20 May 2015 19:48:50 GMT
+	Server: Apache/2.4.7 (Ubuntu)
+	Last-Modified: Wed, 20 May 2015 19:10:18 GMT
+	ETag: "bae1-5178db1739eec"
+	Accept-Ranges: bytes
+	Content-Length: 47841
+	Vary: Accept-Encoding
+	Cache-Control: max-age=3600
+	Expires: Wed, 20 May 2015 20:48:50 GMT
+	Content-Type: text/html
+
+	<!DOCTYPE html>
+	<html lang="en">
+	<head>
+	  <meta charset="utf-8">
+
+	(...)
+
+	</body>
+	</html>
+
+If the requested resource is found on the specified host, it is returned in the body of the response.
+
+cURL
+----
+
+cURL is a tool and library for transferring data with URL syntax, capable of handling various protocols.
+In contrast to telnet, cURL is able to create the HTTP request (or other known protocols) based on the given input.
+It is a command line tool, and can be used in scripts to automate HTTP communication.
+
+In its simplest form, cURL sends a GET request to the given target
+
+.. code-block:: console
+	
+	$ curl localhost:8080/index.html
+	<html>hello there world!</html>
+
+With the ``--request`` or ``-X`` parameter, the method can be specified.
+To include the headers in the output, the ``-i`` can be used, and to only output the headers, use the ``-I`` switch.
+
+.. code-block:: console
+	
+	$ curl -I --request GET http://apache.org/
+	HTTP/1.1 200 OK
+	Date: Wed, 20 May 2015 20:08:43 GMT
+	Server: Apache/2.4.7 (Ubuntu)
+	Last-Modified: Wed, 20 May 2015 19:10:18 GMT
+	ETag: "b962-5179b1a8ef44b"
+	Accept-Ranges: bytes
+	Content-Length: 47458
+	Vary: Accept-Encoding
+	Cache-Control: max-age=3600
+	Expires: Wed, 20 Jun 2015 21:08:43 GMT
+
+cURL outputs a progress meter to the terminal, showing statistics about the operation.
+However, once cURL is about to write data to the terminal, the display of the progress meter is disabled so it may seem as if it was never present.
+The progress meter can still be seen when the output is written ``-o <file>`` or piped ``>`` to a file.
+
+In case the download takes some time (for example downloading a big file) the progress meter can still be seen in the terminal:
+
+.. code-block:: console
+
+	$ curl -O http://localhost/bigfile
```

Comment:
The text above uses `-o <file>`, but the example doesn't say what `-O` means. Is that fine?